content/en/blog/_posts/2021-03-31-kube-state-metrics-goes-v-2-0.md

```diff
@@ -0,0 +1,168 @@
+
+## What is new in v2.0?
+
+So now that we know what kube-state-metrics is, we are excited to announce the next release: kube-state-metrics v2.0! This release was long-awaited and started with an alpha release in September 2020. To ease maintenance we removed tech debt and also adjusted some confusing wording around user-facing flags and APIs. We also removed some metrics that caused unnecessarily high cardinality in Prometheus! For the 2.0 release, we took the time to set up scale and performance testing. This allows us to better understand if we hit any issues in large clusters and also to document resource request recommendations for your clusters. In this release (and v1.9.8) container builds providing support for multiple architectures were introduced allowing you to run kube-state-metrics on ARM, ARM64, PPC64 and S390x as well!
+
+The following is a list of the most impactful user-facing breaking changes. The full list of 2.0 release changes includes other breaking changes regarding metrics dropped or renamed, along with new metrics and features added, for that see the full [CHANGELOG](https://github.com/kubernetes/kube-state-metrics/blob/master/CHANGELOG.md) file and the end.
+
+So without further ado, here is the list of more noteworthy user-facing breaking changes. A full list of changes, features and bug fixes is available in the changelog at the end of this post.
+* Flag `--namespace` was renamed to `--namespaces`. If you are using that make sure to adjust that first before deploying the latest release.
+* Flag `--collectors` was renamed to `--resources`.
+* Flags `--metric-blacklist` and `--metric-whitelist` were renamed to `--metric-denylist` and `--metric-allowlist`.
+* Flag `--metric-labels-allowlist` allows you to specify a list of Kubernetes labels that get turned into the dimensions of the `kube_<resource-name>_labels` metrics. By default, the metric contains only name and namespace labels.
+* All metrics with a prefix of `kube_hpa_*` were renamed to `kube_horizontalpodautoscaler_*`.
+* Kubernetes labels were converted to snake case.
+* If you are importing kube-state-metrics as a library, we have updated our go module path to `k8s.io/kube-state-metrics/v2`
+* All deprecated stable metrics were removed.
+* quay.io/coreos/kube-state-metrics images will no longer be updated. k8s.gcr.io/kube-state-metrics/kube-state-metrics is the new canonical location.
+
+## Found a problem?
+
+Thanks to all our users for testing so far and thank you to all our contributors for your issue reports as well as code and documentation changes! We have cut 2.0.0 X weeks ago, so far we addressed any problems and got reports from users that all is good. If you find any problems, we the maintainers are more than happy to look into them, so please report them by opening a [GitHub issue](https://github.com/kubernetes/kube-state-metrics/issues).
+
+## Full changelog
+
```

Comment:
The formatting here is a bit off - maybe use an [unordered list](https://commonmark.org/help/tutorial/06-lists.html)?

If your editor lets you turn those issue (or PR?) numbers into hyperlinks in one go, I recommend that to.

**Alternatively**: don't include the changelog here, hyperlink to the full changelog at the end of the list of highlights.

Reply:
What would you say is more readable, including a full list with unordered or linking?